### PR TITLE
fix #35: possibly incomplete list of pass by value violations

### LINKED ISSUE #35: pass-by-value on Eigen types may cause alignment issues
Per Eigen docs, passing eign types by value may cause alignment issues and unexplained crashes:

http://eigen.tuxfamily.org/dox-devel/group__TopicPassingByValue.html

At least two instance of this exist in the project.

`./cartographer/transform/transform.h`

```
template <typename T>
Eigen::Matrix<T, 3, 1> RotationQuaternionToAngleAxisVector( Eigen::Quaternion<T> quaternion) 
```
 and 
```
template <typename T>
Eigen::Quaternion<T> AngleAxisVectorToRotationQuaternion(Eigen::Matrix<T, 3, 1> angle_axis)

```

There may be others.

### PR DESCRIPTION
**Pull request by catskul**
Fixes #35: 2 detected instance of this alignment rule violation. There may be more.

Builds and passes tests.

### CODE REVIEW

**Comment by catskul** [comment]
CLA likely resolved by tomorrow 2016-10-07.

File: cartographer/transform/transform.h [hunk 1]
```diff
@@ -51,7 +51,8 @@ T GetYaw(const Rigid3<T>& transform) {
 // rotation as the given 'quaternion'.
 template <typename T>
 Eigen::Matrix<T, 3, 1> RotationQuaternionToAngleAxisVector(
-    Eigen::Quaternion<T> quaternion) {
+    const Eigen::Quaternion<T>& _quaternion) {
```
Comment (wohe) [inline cartographer/transform/transform.h]: This code base follows the [Google C++ style](https://google.github.io/styleguide/cppguide.html), formatting is easiest done by installing clang-format and invoking, e.g., `clang-format -i -style=Google cartographer/transform/transform.h`.

Here: Revert the argument name to 'quaternion', and rather make the local name more descriptive, e.g.:
`const Eigen::Quaternion<T> normalized = quaternion.normalized();`

Comment (catskul) [inline cartographer/transform/transform.h]: roger. will do.

**Comment by catskul** [comment]
CLA now signed for Carnegie Robotics

File: cartographer/transform/transform.h [hunk 2]
```diff
@@ -51,31 +51,31 @@ T GetYaw(const Rigid3<T>& transform) {
 // rotation as the given 'quaternion'.
 template <typename T>
 Eigen::Matrix<T, 3, 1> RotationQuaternionToAngleAxisVector(
-    Eigen::Quaternion<T> quaternion) {
-  quaternion.normalize();
+    const Eigen::Quaternion<T>& quaternion) {
+  auto normalized_quat = quaternion.normalized();
```
Comment (wohe) [inline cartographer/transform/transform.h]: Consistency: Using `auto` with Eigen is problematic ([expression templates](https://en.wikipedia.org/wiki/Expression_templates), see https://eigen.tuxfamily.org/dox/TopicPitfalls.html) so it is best avoided. Prefer `Eigen::Quaternion<T> normalized_quaternion = ...`

Style: Avoid abbreviatons. `normalized_quaternion`?

Comment (catskul) [inline cartographer/transform/transform.h]: Roger. Fixed.

File: cartographer/transform/transform.h [hunk 3]
```diff
@@ -51,31 +51,31 @@ T GetYaw(const Rigid3<T>& transform) {
...
-    quaternion.x() *= T(-1.);
-    quaternion.y() *= T(-1.);
-    quaternion.z() *= T(-1.);
+    normalized_quaternion.w() *= T(-1.);
+    normalized_quaternion.x() *= T(-1.);
+    normalized_quaternion.y() *= T(-1.);
+    normalized_quaternion.z() *= T(-1.);
   }
-  // We convert the quaternion into a vector along the rotation axis with
+  // We convert the normalized_quaternion into a vector along the rotation axis with
```
Comment (wohe) [inline cartographer/transform/transform.h]: Please run `clang-format`.

Comment (catskul) [inline cartographer/transform/transform.h]: Done.

**Comment by wohe** [comment]
@catskul Thanks for the contribution. Can you update the branch? Once it's updated and passes CI, we can merge it.

**Comment by catskul** [comment]
Done.

**Pull request merged**